src/main/java/bisq/core/offer/OfferPayload.java

```diff
@@ -369,4 +368,48 @@ public PublicKey getOwnerPubKey() {
     public String getCurrencyCode() {
         return CurrencyUtil.isCryptoCurrency(getBaseCurrencyCode()) ? getBaseCurrencyCode() : getCounterCurrencyCode();
     }
+
+    @Override
+    public String toString() {
+        return "OfferPayload{" +
+                "\n     id='" + id + '\'' +
+                ",\n     date=" + date +
+                ",\n     ownerNodeAddress=" + ownerNodeAddress +
+                ",\n     pubKeyRing=" + pubKeyRing +
+                ",\n     direction=" + direction +
+                ",\n     price=" + price +
+                ",\n     marketPriceMargin=" + marketPriceMargin +
+                ",\n     useMarketBasedPrice=" + useMarketBasedPrice +
+                ",\n     amount=" + amount +
+                ",\n     minAmount=" + minAmount +
+                ",\n     baseCurrencyCode='" + baseCurrencyCode + '\'' +
+                ",\n     counterCurrencyCode='" + counterCurrencyCode + '\'' +
+                ",\n     arbitratorNodeAddresses=" + arbitratorNodeAddresses +
+                ",\n     mediatorNodeAddresses=" + mediatorNodeAddresses +
+                ",\n     paymentMethodId='" + paymentMethodId + '\'' +
+                ",\n     makerPaymentAccountId='" + makerPaymentAccountId + '\'' +
+                ",\n     offerFeePaymentTxId='" + offerFeePaymentTxId + '\'' +
+                ",\n     countryCode='" + countryCode + '\'' +
+                ",\n     acceptedCountryCodes=" + acceptedCountryCodes +
+                ",\n     bankId='" + bankId + '\'' +
+                ",\n     acceptedBankIds=" + acceptedBankIds +
+                ",\n     versionNr='" + versionNr + '\'' +
+                ",\n     blockHeightAtOfferCreation=" + blockHeightAtOfferCreation +
+                ",\n     txFee=" + txFee +
+                ",\n     makerFee=" + makerFee +
+                ",\n     isCurrencyForMakerFeeBtc=" + isCurrencyForMakerFeeBtc +
+                ",\n     buyerSecurityDeposit=" + buyerSecurityDeposit +
+                ",\n     sellerSecurityDeposit=" + sellerSecurityDeposit +
+                ",\n     maxTradeLimit=" + maxTradeLimit +
+                ",\n     maxTradePeriod=" + maxTradePeriod +
+                ",\n     useAutoClose=" + useAutoClose +
+                ",\n     useReOpenAfterAutoClose=" + useReOpenAfterAutoClose +
+                ",\n     lowerClosePrice=" + lowerClosePrice +
+                ",\n     upperClosePrice=" + upperClosePrice +
+                ",\n     isPrivateOffer=" + isPrivateOffer +
+                ",\n     hashOfChallenge='" + hashOfChallenge + '\'' +
+                ",\n     extraDataMap=" + extraDataMap +
+                ",\n     protocolVersion=" + protocolVersion +
+                "\n}";
+    }
```

Comment:
This was painful, could @ToString.Exclude be used instead?

Reply:
I have a custom template for toString:
If you do ctrl + enter you can generate toString. There in the opening window is a settings button right and there are the templates...
The Lombok toString is not editable and for readability I prefer linebreaks, as well there are some type specific behavior (bytes get printed as hex).

public java.lang.String toString() {
#if ( $members.size() > 0 )
#set ( $i = 0 )
    return "$classname{" +
#foreach( $member in $members )
#if ( $i == 0 )
    "\n     ##
#else
    ",\n     ##
#end
#if ( $member.objectArray )
#if ($java_version < 5)
$member.name=" + ($member.accessor == null ? null : bisq.common.util.Utilities.bytesAsHexString($member.accessor)) +
#else
$member.name=" + bisq.common.util.Utilities.bytesAsHexString($member.accessor) +
#end
#elseif ( $member.primitiveArray && $java_version >= 5)
$member.name=" + bisq.common.util.Utilities.bytesAsHexString($member.accessor) +
#elseif ( $member.string )
$member.name='" + $member.accessor + '\'' +
#else
$member.name=" + $member.accessor +
#end
#set ( $i = $i + 1 )
#end
   #if ( $class.hasSuper )
   "\n} " + super.toString();
   #else
   "\n}";
   #end
#else
   #if ( $class.hasSuper )
   return "$classname{} " + super.toString();
   #else
   return "$classname{}";
   #end
#end
}